app/helpers/comments_helper.rb

```diff
@@ -105,22 +105,21 @@ def get_pingable(thread)
   # @return [Array(Boolean, String)] 2-tuple: boolean indicating if the user is rate-limited, and a string containing
   #   a rate limit message if the user is rate-limited.
   def comment_rate_limited?(user, post, create_audit_log: true)
-    # Comments created by the current user in the last 24 hours, excluding comments on own posts and responses to them.
-    recent_comments = Comment.where(created_at: 24.hours.ago..DateTime.now, user: user).where \
-                             .not(post: Post.includes(:parent).where(parents_posts: { user_id: user.id })) \
-                             .where.not(post: Post.where(user_id: user.id)).count
-    max_comments_per_day = SiteSetting[user.privilege?('unrestricted') ? 'RL_Comments' : 'RL_NewUserComments']
+    comments_count = user.recent_comments_count
+    max_comments_per_day = user.max_comments_per_day
 
-    if post.user_id != user.id && post.parent&.user_id != user.id
+    if user.owns_post_or_parent?(post)
+      [false, nil]
+    else
       if !user.privilege?('unrestricted')
         message = 'As a new user, you can only comment on your own posts and on answers to them.'
         if create_audit_log
           AuditLog.rate_limit_log(event_type: 'comment', related: post, user: user,
                                   comment: "limit: #{max_comments_per_day}")
         end
         [true, message]
-      elsif recent_comments >= max_comments_per_day
-        message = "You have used your daily comment limit of #{recent_comments} comments. Come back tomorrow to " \
+      elsif comments_count >= max_comments_per_day
+        message = "You have used your daily comment limit of #{comments_count} comments. Come back tomorrow to " \
```

Comment:
Don't we need to return `true` and `message` in here?

Reply:
Not sure what you mean by that? Both braches that are supposed to return a (true, string) tuple do retun it? Nothing about the original logic has changed there

Reply:
Oh, it was collapsed away and I missed it, oops!